client/src/components/WeatherCard.jsx

```diff
@@ -0,0 +1,30 @@
+import React from "react";
+
+const WeatherCard = ({ weatherData }) => {
+  return (
+    <div className="weatherDisplay">
+      <div className="icon">
+        <h3>
+          <span>{weatherData.city}</span>
+        </h3>
+        {weatherData.icon && (
```

Comment:
So `icon` can be undefined? Do you know when it could be undefined?

Reply:
Maybe if for some reason Open Weather Map has an error and there is no image associated with the data, `icon` could be undefined.